client/order/cancel-confirm-modal/index.js

```diff
@@ -0,0 +1,83 @@
+/* global jQuery */
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import interpolateComponents from 'interpolate-components';
+import { useState } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import ConfirmationModal from 'wcpay/components/confirmation-modal';
+
+const CancelConfirmationModal = ( { originalOrderStatus } ) => {
+	const [
+		isCancelConfirmationModalOpen,
+		setIsCancelConfirmationModalOpen,
+	] = useState( true );
+
+	const closeModal = () => {
+		setIsCancelConfirmationModalOpen( false );
+	};
+
+	const doNotCancel = () => {
+		jQuery( '#order_status' ).val( originalOrderStatus ).change();
```

Comment:
`jQuery( '#order_status' ).val( originalOrderStatus ).change();`
can only be replaced by the following lines, let me know what you think
`
const orderStatusElement = document.getElementById( 'order_status' );
orderStatusElement.value = originalOrderStatus;
orderStatusElement.dispatchEvent( new Event( 'change' ) );
`

Reply:
>Though it would be good idea to double check if we are using jQuery for legacy reasons.

Not needed anymore, we stopped IE support ~6 months ago, so we can embrace vanilla now 🙂 

>Note for Frontend Guild: Can we start the discussion to get rid of jQuery and promote using native JS, jQuery is a bloated library. Possibly, we can reduce bundle size by getting rid of that.

I'd love to 🤔 Although I'm not sure if rewriting all the working code, just to get rid of jQuery is something we should do soon. But start writing new code in vanilla sounds like a good option.

And no issue with the bundle, we're using the global jQuery included in WP.

>can only be replaced by the following lines, let me know what you think

I'd replace `getElementById ` by `querySelector` it's faster, more consistent, and allow using any selector. But it's just a nitpick 🙂

Reply:
Done